daml/Marketplace/Custody.daml

```diff
@@ -120,6 +138,42 @@ template Service
           TransferDepositRequest{..} <- fetchAndArchive transferDepositRequestCid
           exerciseByKey @AssetSettlementRule accountId AssetSettlement_Transfer with ..
 
+      nonconsuming Lifecycle : [ContractId AssetDeposit]
+        with
+          lifecycleRequestCid: ContractId LifecycleRequest
+          safekeepingDepositCid: ContractId AssetDeposit
+          fixings: Map Text (Map Date Decimal) --TODO these should be looked up from a contract on-ledger, not provided by the custodian!
+        do
+          t <- toDateUTC <$> getTime
+          LifecycleRequest{assetDepositCid, postAssetVersion} <- fetch lifecycleRequestCid
+          assetDeposit <- fetch assetDepositCid
+          (_, description) <- fetchByKey @AssetDescription assetDeposit.asset.id
+          (_, description') <- fetchByKey @AssetDescription (assetDeposit.asset.id with version = postAssetVersion)
+          let payout: Id -> Decimal -> Date -> Update (ContractId AssetDeposit)
+              payout asset qty dt = do
+                payoutDepositCid :: _ <- exercise safekeepingDepositCid $ AssetDeposit_Split [qty]
+                exerciseByKey @AssetSettlementRule description.safekeepingAccountId $
```

Comment:
I think we still need to call `AssetDeposit_AddObserver` here, otherwise the settlement is missing the target signatory. The reason it works in the `Tests/BinaryOption.daml` is that we add it manually (but don't remove it) when transferring the option to Bob.